toolkit/tools/internal/resources/verity-signature/90mountbootpartition/mountbootpartition-generator.sh

```diff
@@ -1,81 +1,41 @@
 #!/bin/sh
 
-set -x
-set -e
+echo "mountbootpartition-generator.sh: Running" > /dev/kmsg
 
-echo "Running mountbootpartition-generator.sh" > /dev/kmsg
+command -v getarg > /dev/null || . /lib/dracut-lib.sh
 
-# type getarg > /dev/null 2>&1 || . /lib/dracut-lib.sh
+# systemd provides a directory to place the generated unit files.
+outputDir="$1"
 
-function updateVeritySetupUnit () {
-    systemdDropInDir=/etc/systemd/system
-    verityDropInDir=$systemdDropInDir/systemd-veritysetup@usr.service.d
+bootPartitionUuid=$(getarg pre.verity.mount) || bootPartitionUuid=""
+if [[ -z "$bootPartitionUuid" ]]; then
+    echo "mountbootpartition-generator.sh: cmdline arg pre.verity.mount is not specified" > /dev/kmsg
+    exit 0
+fi
 
-    mkdir -p $verityDropInDir
-    verityConfiguration=$verityDropInDir/verity-azl-extension.conf
+echo "mountbootpartition-generator.sh: Adding boot.mount" > /dev/kmsg
 
-    cat <<EOF > $verityConfiguration
-[Unit]
-After=bootmountmonitor.service
-Requires=bootmountmonitor.service
-EOF
-
-    chmod 644 $verityConfiguration
-    chown root:root $verityConfiguration
-}
-
-# -----------------------------------------------------------------------------
-function createBootPartitionMonitorScript () {
-    local bootPartitionMonitorCmd=$1
-    local semaphorefile=$2
-
-    cat <<EOF > $bootPartitionMonitorCmd
-#!/bin/sh
-while [ ! -e "$semaphorefile" ]; do
-    echo "Waiting for $semaphorefile to exist..."
-    sleep 1
-done    
-EOF
-    chmod +x $bootPartitionMonitorCmd
-}
+bootMountMonitorUnitFile="$outputDir/boot.mount"
 
-# -----------------------------------------------------------------------------
-function createBootPartitionMonitorUnit() {
-    local bootPartitionMonitorCmd=$1
-
-    bootMountMonitorName="bootmountmonitor.service"
-    systemdDropInDir=/etc/systemd/system
-    bootMountMonitorDir=$systemdDropInDir
-    bootMountMonitorUnitFile=$bootMountMonitorDir/$bootMountMonitorName
-
-    cat <<EOF > $bootMountMonitorUnitFile
+cat <<EOF > $bootMountMonitorUnitFile
 [Unit]
-Description=bootpartitionmounter
-DefaultDependencies=no
-
-[Service]
-Type=oneshot
-ExecStart=$bootPartitionMonitorCmd
-RemainAfterExit=yes
-
-[Install]
-WantedBy=multi-user.target
+Description=/boot
+RequiresMountsFor=/dev/disk/by-uuid/$bootPartitionUuid
```

Comment:
Does `RequiresMountsFor=` have the same effect as the `After=dev-disk-by\\x2duuid-${bootPartitionUuid//-/\\x2d}.device` directive? They both take `partuuid`.